### PR TITLE
📝Documentation as static website

### PR DESCRIPTION
**Pull request by nadvitek**
Added GitHub Action to build ACKategories documentation and deploy it as a static website with Github Pages.

### CODE REVIEW

File: .github/workflows/docbuild.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  cancel-in-progress: true
+
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      # Must be set to this for deploying to GitHub Pages
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: macos-12
```
Comment (olejnjak) [inline .github/workflows/docbuild.yml:24]: I would prefer running on macOS 13 as all other jobs do.

File: .github/workflows/docbuild.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
+      - name: Build DocC
+        run: |
+          xcodebuild docbuild -scheme ACKategories \
```
Comment (olejnjak) [inline .github/workflows/docbuild.yml:30]: As long as you use xcodebuild I would call the load xcode version action as all other actions do to make sure you use correct Xcode version

File: .github/workflows/docbuild.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      # Must be set to this for deploying to GitHub Pages
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: macos-12
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
```
Comment (olejnjak) [inline .github/workflows/docbuild.yml:27]: Would be good to use latest checkout version - v4

**Review by olejnjak — CHANGES_REQUESTED** [review]

**Review by olejnjak — APPROVED** [review]

**Review by IgorRosocha — APPROVED** [review]
LGTM, good job. 🙌

**Pull request merged**